java-client/build.gradle.kts

```diff
@@ -369,5 +372,17 @@ if (JavaVersion.current() >= JavaVersion.VERSION_11) {
     testClassesDirs += java11.output.classesDirs
     classpath = sourceSets["java11"].runtimeClasspath
   }
- 
+} else if (jdk8compatibility == true) {
+    java {
```

Comment:
@dblock @VachaShah curious what you think about this approach: we cannot run Gradle build using JDK-8 but we could use toolchains to run tests under JDK-8.